### ISSUE TITLE
Switching from Kaae to Sentinl

### ISSUE THREAD

**Issue by kayrus** [sirensolutions/sentinl#92]
Just switched my configuration from Kaae to Sentinl. And it shows nothing 
![screenshot_20170113_162100](https://cloud.githubusercontent.com/assets/4047483/21935081/4a8af8b4-d9ac-11e6-90c9-e83cd80f375e.png)

Previous config:

```json
{
    "es": {
        "alarm_index": "watcher_alarms",
        "default_index": "watcher",
        "timefield": "time",
        "type": "watch"
    },
... ... ...
```

New config:

```yaml
sentinl:
  es:
    timefield: 'time'
    default_index: watcher
    type: watch
    alarm_index: watcher_alarms
... ... ...
```

Docker image build log:

```
Step 1 : FROM mhart/alpine-node:5
 ---> 89775bcb2ef1
Step 2 : EXPOSE 5601
 ---> Running in 3f3203eb955f
 ---> 76bc0fc25fee
Removing intermediate container 3f3203eb955f
Step 3 : ENV KIBANA_VERSION 4.6.1
 ---> Running in 4b803b05aa94
 ---> 9d8971f417b0
Removing intermediate container 4b803b05aa94
Step 4 : RUN apk add --update curl ca-certificates sudo &&   ( curl -Lskj https://download.elastic.co/kibana/kibana/kibana-${KIBANA_VERSION}-linux-x86_64.tar.gz |   gunzip -c - | tar xf - ) &&   mv /kibana-${KIBANA_VERSION}-linux-x86_64 /kibana-linux-x86_64 &&   rm -rf /kibana-linux-x86_64/node &&   apk del curl
 ---> Running in 9d7a3e3f1e62
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/community/x86_64/APKINDEX.tar.gz
(1/5) Installing ca-certificates (20160104-r4)
(2/5) Installing libssh2 (1.7.0-r0)
(3/5) Installing libcurl (7.52.1-r0)
(4/5) Installing curl (7.52.1-r0)
(5/5) Installing sudo (1.8.16-r0)
Executing busybox-1.24.2-r9.trigger
Executing ca-certificates-20160104-r4.trigger
OK: 8 MiB in 18 packages
(1/3) Purging curl (7.52.1-r0)
(2/3) Purging libcurl (7.52.1-r0)
(3/3) Purging libssh2 (1.7.0-r0)
Executing busybox-1.24.2-r9.trigger
OK: 8 MiB in 15 packages
 ---> f08de0b57462
Removing intermediate container 9d7a3e3f1e62
Step 5 : RUN npm install --save later mustache
 ---> Running in fabf6bb95e8c
npm WARN saveError ENOENT: no such file or directory, open '/package.json'
/
+-- later@1.2.0 
`-- mustache@2.3.0 

npm WARN enoent ENOENT: no such file or directory, open '/package.json'
npm WARN !invalid#1 No description                                                                                      
npm WARN !invalid#1 No repository field.                                                                                
npm WARN !invalid#1 No README data                                                                                      
npm WARN !invalid#1 No license field.                                                                                   
 ---> be819119c470
Removing intermediate container fabf6bb95e8c
Step 6 : RUN /kibana-linux-x86_64/bin/kibana plugin --install sentinl -u https://github.com/sirensolutions/sentinl/archive/tag-4.6.3.tar.gz
 ---> Running in 79391f8c1b5c
Installing sentinl
Attempting to transfer from https://github.com/sirensolutions/sentinl/archive/tag-4.6.3.tar.gz
Transferring unknown number of bytes
Transfer complete
Extracting plugin archive
Extraction complete
Optimizing and caching browser bundles...
Plugin installation complete
 ---> 35642e3fb840
Removing intermediate container 79391f8c1b5c
Step 7 : RUN /kibana-linux-x86_64/bin/kibana plugin --install heatmap -u https://github.com/stormpython/heatmap/archive/1.0.0.zip
 ---> Running in a1032e8a5a64
Installing heatmap
Attempting to transfer from https://github.com/stormpython/heatmap/archive/1.0.0.zip
Transferring unknown number of bytes
Transfer complete
Extracting plugin archive
Extraction complete
Optimizing and caching browser bundles...
Plugin installation complete
 ---> 3384cde9d482
Removing intermediate container a1032e8a5a64
Step 8 : RUN /kibana-linux-x86_64/bin/kibana plugin --install vectormap -u https://github.com/stormpython/vectormap/archive/master.zip
 ---> Running in 34ea64ce6c30
Installing vectormap
Attempting to transfer from https://github.com/stormpython/vectormap/archive/master.zip
Transferring unknown number of bytes
Transfer complete
Extracting plugin archive
Extraction complete
Optimizing and caching browser bundles...
[BABEL] Note: The code generator has deoptimised the styling of "/kibana-linux-x86_64/installedPlugins/vectormap/public/lib/jvectormap/maps/map.asia_mill.js" as it exceeds the max of "100KB".
[BABEL] Note: The code generator has deoptimised the styling of "/kibana-linux-x86_64/installedPlugins/vectormap/public/lib/jvectormap/maps/map.europe_mill.js" as it exceeds the max of "100KB".
[BABEL] Note: The code generator has deoptimised the styling of "/kibana-linux-x86_64/installedPlugins/vectormap/public/lib/jvectormap/maps/map.world_mill.js" as it exceeds the max of "100KB".
Plugin installation complete
 ---> 05c6051c305b
Removing intermediate container 34ea64ce6c30
Step 9 : RUN rm -rf /kibana-linux-x86_64/config && ln -s /etc/kibana /kibana-linux-x86_64/config
 ---> Running in bf91ccd2de77
 ---> 21be1e0e3b4d
Removing intermediate container bf91ccd2de77
Step 10 : COPY run.sh /
 ---> 7f732f49f103
Removing intermediate container 5b4e94db85df
Step 11 : CMD /run.sh
 ---> Running in 2b5f999e370d
 ---> 46c627536a24
Removing intermediate container 2b5f999e370d
Successfully built 46c627536a24
```

**Comment by lmangani** [comment]
Mapping changed to support reporting and other features - are there any errors on the kibana logs in verbose mode? Ideally, let Sentinl recreate the indices and then either re-create or plain import the existing watchers.

**Comment by kayrus** [comment]
@lmangani there is nothing about incompatible kaae/sentinl indexes. Will try to recreate them from the scratch.

**Comment by lmangani** [comment]
any errors in the browser console?

**Comment by kayrus** [comment]
```
GET https://mysite.com/api/sentinl/example 404 ()
```

**Comment by kayrus** [comment]
I use sentinl 4.6.3 tag.

**Comment by lmangani** [comment]
what's that? I mean if there are any errors in your console browser while loading the Kibana App :)

**Comment by kayrus** [comment]
Yep, Chrome complains that it can not load `https://mysite.com/api/sentinl/example`

**Comment by kayrus** [comment]
Stopped kibana, removed old indexes, started. New indexes were not created.

**Comment by kayrus** [comment]
Kibana logs:

```
{"type":"log","@timestamp":"2017-01-13T15:51:09Z","tags":["status","plugin:heatmap@0.1.0","info"],"pid":7,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-01-13T15:51:09Z","tags":["status","plugin:kibana@1.0.0","info"],"pid":7,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-01-13T15:51:10Z","tags":["status","plugin:elasticsearch@1.0.0","info"],"pid":7,"state":"yellow","message":"Status changed from uninitialized to yellow - Waiting for Elasticsearch","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-01-13T15:51:10Z","tags":["status","plugin:sentinl@4.6.3","info"],"pid":7,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-01-13T15:51:10Z","tags":["status","plugin:vectormap@0.1.0","info"],"pid":7,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-01-13T15:51:10Z","tags":["status","plugin:kbn_vislib_vis_types@1.0.0","info"],"pid":7,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-01-13T15:51:11Z","tags":["status","plugin:markdown_vis@1.0.0","info"],"pid":7,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-01-13T15:51:11Z","tags":["status","plugin:metric_vis@1.0.0","info"],"pid":7,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-01-13T15:51:11Z","tags":["status","plugin:spyModes@1.0.0","info"],"pid":7,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-01-13T15:51:11Z","tags":["status","plugin:elasticsearch@1.0.0","info"],"pid":7,"state":"green","message":"Status changed from yellow to green - Kibana index ready","prevState":"yellow","prevMsg":"Waiting for Elasticsearch"}
{"type":"log","@timestamp":"2017-01-13T15:51:11Z","tags":["status","plugin:elasticsearch@1.0.0","error"],"pid":7,"state":"red","message":"Status changed from green to red - Unknown config key: elasticsearch.plugins","prevState":"green","prevMsg":"Kibana index ready"}
{"type":"log","@timestamp":"2017-01-13T15:51:11Z","tags":["status","plugin:statusPage@1.0.0","info"],"pid":7,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-01-13T15:51:11Z","tags":["status","plugin:table_vis@1.0.0","info"],"pid":7,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-01-13T15:51:11Z","tags":["listening","info"],"pid":7,"message":"Server running at http://0.0.0.0:5601"}
{"type":"log","@timestamp":"2017-01-13T15:51:14Z","tags":["status","plugin:elasticsearch@1.0.0","info"],"pid":7,"state":"green","message":"Status changed from red to green - Kibana index ready","prevState":"red","prevMsg":"Unknown config key: elasticsearch.plugins"}
```

**Comment by kayrus** [comment]
@lmangani nope. I even don't know what causes this issue.

**Comment by lmangani** [comment]
If you remove sentinl, does the error disappear? What happens if you remove the custom yml?

**Comment by kayrus** [comment]
Here is the log of kaae starup:

```
"type":"log","@timestamp":"2017-02-07T11:58:47Z","tags":["status","plugin:heatmap@0.1.0","info"],"pid":6,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-02-07T11:58:47Z","tags":["status","plugin:kibana@1.0.0","info"],"pid":6,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-02-07T11:58:48Z","tags":["status","plugin:elasticsearch@1.0.0","info"],"pid":6,"state":"yellow","message":"Status changed from uninitialized to yellow - Waiting for Elasticsearch","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-02-07T11:58:48Z","tags":["status","info","KaaE"],"pid":6,"message":"KaaE Initializing"}
{"type":"log","@timestamp":"2017-02-07T11:58:50Z","tags":["status","info","KaaE"],"pid":6,"message":"Core Index check..."}
{"type":"log","@timestamp":"2017-02-07T11:58:50Z","tags":["status","info","KaaE"],"pid":6,"message":"Alarm Index check..."}
{"type":"log","@timestamp":"2017-02-07T11:58:50Z","tags":["status","plugin:kaae@0.0.9","info"],"pid":6,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-02-07T11:58:51Z","tags":["status","plugin:vectormap@0.1.0","info"],"pid":6,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-02-07T00:00:00Z","tags":["status","info","KaaE"],"pid":6,"message":"Core Index response"}
{"type":"log","@timestamp":"2017-02-07T00:00:00Z","tags":["status","info","KaaE"],"pid":6,"message":"Alarm Index response"}
{"type":"log","@timestamp":"2017-02-07T11:58:51Z","tags":["status","plugin:kbn_vislib_vis_types@1.0.0","info"],"pid":6,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-02-07T11:58:51Z","tags":["status","plugin:markdown_vis@1.0.0","info"],"pid":6,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-02-07T11:58:51Z","tags":["status","plugin:metric_vis@1.0.0","info"],"pid":6,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-02-07T11:58:51Z","tags":["status","plugin:elasticsearch@1.0.0","info"],"pid":6,"state":"green","message":"Status changed from yellow to green - Kibana index ready","prevState":"yellow","prevMsg":"Waiting for Elasticsearch"}
{"type":"log","@timestamp":"2017-02-07T11:58:51Z","tags":["status","plugin:spyModes@1.0.0","info"],"pid":6,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-02-07T11:58:51Z","tags":["status","plugin:statusPage@1.0.0","info"],"pid":6,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-02-07T11:58:51Z","tags":["status","plugin:table_vis@1.0.0","info"],"pid":6,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-02-07T11:58:51Z","tags":["listening","info"],"pid":6,"message":"Server running at http://0.0.0.0:5601"}
```

There is no `Unknown config` log.

**Comment by lmangani** [comment]
You're running the old plugin. The new one produces logs as SENTINL

**Comment by kayrus** [comment]
@lmangani what do you mean? You asked me to start kibana without SENTINL. I did this. Of course it won't have "SENTINL" logs :)

**Comment by lmangani** [comment]
@kayrus that's not what i meant. If you remove SENTINL and you still see KAAE logs, you are running the old plugin as well. You need to remove it, drop the .watcher* tables and reinstall SENTINL, I guess.

**Comment by kayrus** [comment]
I don't have a container without SENTINL or KAAE. Thus I just started KAAE. You wanted to check whether the `Unknown config key: elasticsearch.plugins` log will appear. It won't without SENTINL plugin.

As for the `.watcher*` tables. They were removed as well.

**Comment by lmangani** [comment]
I'll try to re-read this later on, I can't recall issues months apart but this being said, it might be a good day to try out Kibi (where the plugin is most tested) and see if you can go back to Kibana afterwards ;)

**Comment by kayrus** [comment]
@lmangani I've just noticed new `tag-4.6.3-1` tag. I've installed this version and it works and shows user interface.

As for Kibi, it look fancy, but thanks. I'll postpone it. It is hard to educate people to work with the new project.

**Comment by lmangani** [comment]
@kayrus glad it works. As of Kibi, that's a shame as there is zero learning curve coming from Kibana, and you can offer your users Dashboards, joins and other key visualizers and extensions in full compatibility mode. Also you can run both in parallel without any disturabance as they use different indices for settings.
